### PR TITLE
Add iOS Support

### LINKED ISSUE #2: Need iOS help?
Hi @YoussefKababe,

Awesome work on this plugin! I have an immediate need for banner and interstitial on Android __and__ iOS. I'm newer to Flutter and am still researching PlatformViews however I have experience in iOS.  Are you open to PR's?

### PR DESCRIPTION
**Pull request by kmcgill88**
I did my best to keep the Swift implementation in sync with @YoussefKababe's Kotlin code. Not everything lines up exactly.

1. I ended up registering 3 plugins. `admob_flutter` (the default), `AdmobInterstitialPlugin` and `AdmobRewardPlugin`.
1. I'm not sure unique `id`'s are required Interstitial and Reward but I followed suit for now. I think this should be revisited since Reward is a singleton in Swift and Interstitial's are one use and done.

Thanks for reviewing!

Closes #2

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -233,3 +235,62 @@ class _MyAppState extends State<MyApp> {
...
+*/
+
+String getAppId() {
+  if (Platform.isIOS) {
+    return 'ca-app-pub-3940256099942544~1458002511';
+  } else if (Platform.isAndroid) {
+    return 'ca-app-pub-3940256099942544~3347511713';
+  }
+  return null;
+}
```
Comment (kmcgill88) [inline example/lib/main.dart]: All test `id`'s are different per platform. I was getting weird behavior's when I mixed and Android App Id with and iOS ad unit id. This was a simple fix. I'm open to cleaner suggestions.

File: ios/Classes/AdmobBanner.swift [hunk 2]
```diff
@@ -1,8 +1,134 @@
...
+    func adView(_ bannerView: GADBannerView, didFailToReceiveAdWithError error: GADRequestError) {
+         channel.invokeMethod("failedToLoad", arguments: error)
+    }
+    
+    /// Tells the delegate that a full screen view will be presented in response to the user clicking on
+    /// an ad. The delegate may want to pause animations and time sensitive interactions.
+    func adViewWillPresentScreen(_ bannerView: GADBannerView) {
+        channel.invokeMethod("clicked", arguments: nil)
+        channel.invokeMethod("opened", arguments: nil)
+    }
```
Comment (kmcgill88) [inline ios/Classes/AdmobBanner.swift]: This is one example of iOS not matching up completely with Android. Combining these callbacks to Flutter made sense to me. At least all callbacks will be fired on the Flutter side. @YoussefKababe do you see any problem with this?

File: ios/Classes/AdmobBanner.swift [hunk 3]
```diff
@@ -1,8 +1,134 @@
...
+    
+    // TODO: not sure this exists on iOS.
+    // channel.invokeMethod("impression", null)
```
Comment (kmcgill88) [inline ios/Classes/AdmobBanner.swift]: Does anyone know if this exists on iOS?

File: ios/Classes/SwiftAdmobFlutterPlugin.swift [hunk 4]
```diff
@@ -1,14 +1,50 @@
...
-    result("iOS " + UIDevice.current.systemVersion)
+    switch call.method {
+    case "initialize":
+        GADMobileAds.sharedInstance().start { (initializationStatus: GADInitializationStatus) in
+            print("initializationStatus: \(initializationStatus)")
+        }
+        break
+    case "getPlatformVersion":
+        result("iOS " + UIDevice.current.systemVersion)
+        break
```
Comment (kmcgill88) [inline ios/Classes/SwiftAdmobFlutterPlugin.swift]: @YoussefKababe I don't believe this is used. I'm guessing it's left over from boilerplate templates. If so, I will remove it.

File: ios/admob_flutter.podspec [hunk 5]
```diff
@@ -3,19 +3,22 @@
...
+  s.summary          = 'Admob plugin that shows banner ads using native platform views.'
   s.description      = <<-DESC
-A new flutter plugin project.
+Admob plugin that shows banner ads using native platform views.
                        DESC
-  s.homepage         = 'http://example.com'
+  s.homepage         = 'https://github.com/YoussefKababe/admob_flutter'
   s.license          = { :file => '../LICENSE' }
-  s.author           = { 'Your Company' => 'email@example.com' }
+  s.author           = { 'admob_flutter' => 'email@admobflutter.fake' }
```
Comment (kmcgill88) [inline ios/admob_flutter.podspec]: I guess I could put my name here? 👨‍🎨

File: README.md [hunk 6]
```diff
@@ -16,18 +16,19 @@ This is an early version of the plugin, my primary goal was to overcome the bann
 
 **Currently working on Android only, but iOS support will be available so soon.**
```
Comment (c4talyst) [inline README.md]: You can update this line now :)

Comment (kmcgill88) [reply README.md]: @c4talyst thanks for looking through this. Most of it was written sleep deprived.

Comment (c4talyst) [reply README.md]: I'm no expert here, but will happily review the readme! :) I'll try give the branch a test too

Comment (kmcgill88) [reply README.md]: @c4talyst please do run the branch! I haven't fully flushed out differ banner sizes.

File: ios/Classes/SwiftAdmobFlutterPlugin.swift (hunk 4 above)
Comment (YoussefKababe) [reply ios/Classes/SwiftAdmobFlutterPlugin.swift]: Yes, that's certainly left over from the examples. We can remove it.

File: ios/Classes/AdmobBanner.swift (hunk 2 above)
Comment (YoussefKababe) [reply ios/Classes/AdmobBanner.swift]: @kmcgill88 This looks good to me, can't see a problem with it

File: ios/Classes/SwiftAdmobFlutterPlugin.swift (hunk 4 above)
Comment (kmcgill88) [reply ios/Classes/SwiftAdmobFlutterPlugin.swift]: I'll remove it in a minute. I'm actually pulling in my fork to an app of mine to test things out right now!

**Comment by YoussefKababe** [comment]
@kmcgill88 The above looks good to me! I'm approving the PR, and if you have tested it well and everything seems to work on your side as well, let's merge it! 🎉

**Comment by kmcgill88** [comment]
The only potential issue I've seen is related to banner ads. This only happens occasionally. I wouldn't think the test ads would do this? I'm pulling my fork into my app now. I will try with my real id's.

```
<Google> Cannot find an ad network adapter with the name(s): com.google.DummyAdapter. Remember to link all required ad network adapters and SDKs, and set -ObjC in the 'Other Linker Flags' setting of your build target.
Unsupported value: Error Domain=com.google.admob Code=8 "Request Error: Cannot find an ad network adapter for mediation." UserInfo={NSLocalizedDescription=Request Error: Cannot find an ad network adapter for mediation.} of type GADRequestError
```

**Comment by kmcgill88** [comment]
@YoussefKababe I think this is ready to go! 🚀 Please squash and merge when you have time to publish `0.3.0`.

**Comment by YoussefKababe** [comment]
@kmcgill88 Just merged and releases `0.3.0`. Thank you so much for the awesome work! 🎉 🚀 ❤️

**Pull request merged**